src/he/mss.c

```diff
@@ -81,7 +81,7 @@ he_return_code_t he_internal_clamp_mss(uint8_t *packet, size_t length, uint16_t
   int options_len = tcp_header_len - sizeof(tcp_header_t);
 
   // Check to make sure we're in bounds
-  if(ip_header_len + tcp_header_len > length) {
+  if(ip_header_len + tcp_header_len > length && options_len > 1) {
```

Comment:
I dont understand why is this "&& options_len > 1" needed ?
Can you please update the comment?

Reply:
The `tcp_option_t` is a 2-bytes struct, so if options_len == 1 and the `kind` value > 2, we'll get OOB read at line:141

Reply:
I think the check is wrong then. It should be added to the while() loop check at line 92

Reply:
Fixed.